### PR TITLE
Improve socket template handling.

### PR DESCRIPTION
**Pull request by frankban**
Allow for specifying the full WebSocket URL and not just the
path in socketTemplate.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5003/
Test PASSed.

**Comment by bac** [comment]
Looks 👍 .  No QA.

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -1918,41 +1917,39 @@ YUI.add('juju-gui', function(Y) {
...
-      if (!server || !port) {
-        var apiAddress = this.get('apiAddress').replace('wss://', '');
-        apiAddress = apiAddress.split(':');
-        server = apiAddress[0];
-        port = apiAddress[1];
+    createSocketURL: function(uuid, server, port) {
+      var baseUrl = '';
+      var url = this.get('socketTemplate');
+      var sandbox = this.get('sandbox');
+      if (url[0] === '/' || sandbox) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: Can you add a code comment as to when this would be '/'.

**Comment by hatched** [comment]
:+1: with trivial, no qa yet.

File: jujugui/static/gui/src/app/app.js (hunk 1 above)
Comment (frankban) [inline jujugui/static/gui/src/app/app.js]: Done.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5006/
Test PASSed.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**